foreign_cc/meson.bzl

```diff
@@ -99,46 +99,98 @@ def _create_meson_script(configureParameters):
     root = detect_root(ctx.attr.lib_source)
     data = ctx.attr.data + ctx.attr.build_data
 
-    # Generate a list of arguments for meson
-    options_str = " ".join([
-        "-D{}=\"{}\"".format(key, ctx.attr.options[key])
-        for key in ctx.attr.options
-    ])
+    if attrs.tool_prefix:
+        tool_prefix = "{} ".format(
+            expand_locations_and_make_variables(ctx, attrs.tool_prefix, "tool_prefix", data),
+        )
+    else:
+        tool_prefix = ""
+
+    meson_path = "{tool_prefix}{meson_path}".format(
+        tool_prefix = tool_prefix,
+        meson_path = attrs.meson_path,
+    )
 
-    prefix = "{} ".format(expand_locations_and_make_variables(ctx, attrs.tool_prefix, "tool_prefix", data)) if attrs.tool_prefix else ""
+    target_args = dict(ctx.attr.target_args)
 
-    setup_args_str = " ".join(expand_locations_and_make_variables(ctx, ctx.attr.setup_args, "setup_args", data))
+    # --- TODO: DEPRECATED, delete on a future release ------------------------
+    # Convert the deprecated arguments into the new target_args argument. Fail
+    # if there's a deprecated argument being used together with its new
+    # target_args (e.g. setup_args and a "setup" target_args).
 
-    script.append("{prefix}{meson} setup --prefix={install_dir} {setup_args} {options} {source_dir}".format(
-        prefix = prefix,
-        meson = attrs.meson_path,
-        install_dir = "$$INSTALLDIR$$",
-        setup_args = setup_args_str,
-        options = options_str,
-        source_dir = "$$EXT_BUILD_ROOT$$/" + root,
-    ))
+    deprecated = [
+        ("setup", ctx.attr.setup_args),
+        ("compile", ctx.attr.build_args),
+        ("install", ctx.attr.install_args),
+    ]
+
+    for target_name, args_ in deprecated:
+        if args_ and target_name in target_args:
+            fail("Please migrate '{t}_args' to 'target_args[\"{t}\"]'".format(t = target_name))
+
+        target_args[target_name] = args_
```

Comment:
from my testing, this will override any values I currently have set in `target_args`

Reply:
created https://github.com/bazel-contrib/rules_foreign_cc/pull/1444

Reply:
You are right the code is wrong, but see my comments in the PR.